hiero-enterprise-base/src/test/java/com/openelements/hiero/base/test/SmartContractClientImplTest.java

```diff
@@ -0,0 +1,117 @@
+package com.openelements.hiero.base.test;
+
+import com.hedera.hashgraph.sdk.ContractFunctionResult;
+import com.hedera.hashgraph.sdk.ContractId;
+import com.openelements.hiero.base.FileClient;
+import com.openelements.hiero.base.HieroException;
+import com.openelements.hiero.base.implementation.SmartContractClientImpl;
+import com.openelements.hiero.base.protocol.ProtocolLayerClient;
+import com.openelements.hiero.base.protocol.data.ContractCallRequest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.provider.NullSource;
+import org.junit.jupiter.params.ParameterizedTest;
+
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+public class SmartContractClientImplTest {
+
+    private ProtocolLayerClient mockProtocolLayerClient;
+    private FileClient mockFileClient;
+    private SmartContractClientImpl smartContractClient;
+
+    @BeforeEach
+    public void setUp() {
+        mockProtocolLayerClient = mock(ProtocolLayerClient.class);
+        mockFileClient = mock(FileClient.class);
+        smartContractClient = new SmartContractClientImpl(mockProtocolLayerClient, mockFileClient);
+    }
+
+    @Test
+    void CallContractFunction_success() throws Exception {
+
+        ContractId contractId = ContractId.fromString("0.0.1234");
+        String functionName = "getValue";
+
+        ContractFunctionResult mockFunctionResult = mock(ContractFunctionResult.class);
+        com.openelements.hiero.base.protocol.data.ContractCallResult mockProtocolResult =
+                mock(com.openelements.hiero.base.protocol.data.ContractCallResult.class);
+        when(mockProtocolResult.contractFunctionResult()).thenReturn(mockFunctionResult);
+
+        when(mockProtocolLayerClient.executeContractCallTransaction(any(ContractCallRequest.class)))
+                .thenReturn(mockProtocolResult);
+
+        com.openelements.hiero.base.data.ContractCallResult result =
+                smartContractClient.callContractFunction(contractId, functionName);
+
+        assertNotNull(result);
+
+    }
+
+    @Test
+    void CallContractFunction_exceptionThrown() throws Exception {
```

Comment:
The tests in this class does not really test the API or implementation of the project but jsut creates some mock flows without a real impact of the functionallity of the Api. Sorry but this PR can not be merged